malaria-ios/Managers/AchievementManager.swift

```diff
@@ -0,0 +1,174 @@
+      return result as [Achievement]
+    } catch let error as NSError {
+      Logger.Error("Could not fetch \(error), \(error.userInfo)")
+    }
+    return []
+  }
+  
+  /**
+   Each achievement has it properties that needs to be fulfilled
+   in order to unlock.
+   */
+  
+  private var properties: [AchievementProperty] {
+    let fetchRequest = NSFetchRequest(entityName: "AchievementProperty")
+    
+    do {
+      let result = try context.executeFetchRequest(fetchRequest) as! [AchievementProperty]
+      return result as [AchievementProperty]
+    } catch let error as NSError {
+      Logger.Error("Could not fetch \(error), \(error.userInfo)")
+    }
+    return []
+  }
+  
+  func getAchievements(withTag tag: String) -> [Achievement] {
+    return achievements.filter { $0.tag == tag }
+  }
+  
+  func defineProperty(name: String,
+                      initialValue: Int,
+                      activationMode: String,
+                      value: Int) {
+    
+    // Check for duplicate properties.
+    for property in self.properties where property.name == name {
+        return
+    }
+    
+    let property = NSEntityDescription.insertNewObjectForEntityForName("AchievementProperty", inManagedObjectContext: context) as! AchievementProperty
+    
+    property.name = name
+    property.activation = activationMode
+    property.activationValue = value
+    property.initialValue = initialValue
+    
+    CoreDataHelper.sharedInstance.saveContext(context)
+  }
+  
+  /**
+   Defines an achievement with a name, tag and defined properties name that it will use.
+   
+   Example: defineAchievement("Bounty Hunting",
+   tag: "Games",
+   ["Score 10 Points", "Score 50 Points", "Kill Boss"]
+   */
+  
+  func defineAchievement(name: String,
+                         description: String,
+                         propertyNames: [String],
+                         tag: String) {
+    
+    // Check for duplicate achievements.
+    for achievement in self.achievements {
+      if achievement.name == name {
+        return
+      }
+    }
+    
+    let achievement = NSEntityDescription.insertNewObjectForEntityForName("Achievement", inManagedObjectContext: context) as! Achievement
+    
+    // Check whether the required properties for this achievement exists.
+    let matchingProperties: [AchievementProperty]? =
+      properties.filter { propertyNames.contains($0.name!) }
+    
+    guard matchingProperties != nil else {
+      Logger.Warn("There are no matching properties for the achievement.")
+      return
+    }
+    
+    achievement.properties = Set(matchingProperties!)
+    achievement.name = name
+    achievement.tag = tag
+    achievement.desc = description
+    
+    CoreDataHelper.sharedInstance.saveContext(context)
+  }
+  
+  /// Gets the current value of a `AchievementProperty`.
+  
+  func getValue(propertyName: String) -> Int {
+    return properties.filter { $0.name! == propertyName }
+      .first!
+      .currentValue as! Int
```

Comment:
Create a method that converts the currentValue to Int, in an extension? How could it be called, "getCurrentValue"? I don't understand the idea behind this, but I'm sure this piece of code can be improved.